synapse_invite_checker/invite_checker.py

```diff
@@ -499,10 +499,13 @@ async def _fetch_federation_list(
         # Validate incoming, potentially incomplete or corrupt data
         return FederationList.model_validate_json(jws_verify.payload)
 
-    async def _domain_list_check(self, check: Callable[[str], bool]) -> bool:
-        """Run a `check` against data found on the FederationList"""
+    async def is_domain_allowed(self, domain: str) -> bool:
+        """
+        See if a domain is found on the Federation List. If it was not, refetch the list
+        to try again.
+        """
         fed_list = await self._fetch_federation_list()
-        if check(fed_list):
+        if fed_list.allowed(domain):
```

Comment:
the federation list should be refetched, if the domain is not in the federation list. Also when we are fetching the list to check if the server is an insurance, afaik.

Reply:
But the federation list is not refetched, if an insurance is not in the federation list. But I guess it doesn't matter in the cases, that we care about.